packages/design-system/src/components/Alert/Alert.tsx

```diff
@@ -68,28 +136,33 @@ export class Alert extends React.PureComponent {
     }
   }
 
-  heading() {
-    const Heading = `h${this.props.headingLevel}` || `h2`;
-    if (this.props.heading) {
-      return (
-        <Heading className="ds-c-alert__heading" id={this.headingId}>
-          {this.props.heading}
-        </Heading>
-      );
+  // Alert class properties
+  alertTextRef: any;
+  focusRef: any;
+  headingId: string;
+  eventHeadingText: string;
+
+  heading(): React.ReactElement | void {
+    const { headingLevel = '2', heading } = this.props;
+    const Heading = `h${headingLevel}`;
+    if (heading) {
+      const headingProps = {
+        className: 'ds-c-alert__heading',
+        id: this.headingId,
+      };
+      return React.createElement(Heading, headingProps, heading);
     }
   }
 
-  render() {
+  render(): React.ReactNode {
     const {
       children,
       className,
       autoFocus,
       heading,
-      headingId,
-      headingLevel,
```

Comment:
Can keep the `headingId` and `headingLevel` const here, otherwise the spread operator for `{...alertProps}` will get passed these variables.
Also noticed that the current code is missing `analytics` variable here as well, can you help add that here as well.

Reply:
`for the Analytics event object, should any of those properties be optional?` - good question, these properties would be optional for when applications passes these properties to the Alert component